lib/ansible/modules/net_tools/nsupdate.py

```diff
@@ -172,11 +173,15 @@ class RecordManager(object):
     def __init__(self, module):
         self.module = module
 
-        if module.params['zone'][-1] != '.':
-            self.zone = module.params['zone'] + '.'
+        if module.params['zone'] is None:
+            self.zone = dns.resolver.zone_for_name(self.module.params['record']).to_text()
+            module.params['record'] = module.params['record'][:-len(self.zone)]
```

Comment:
Ok, this shortening of the record parameter is both broken and unnecessary.

I assume it is to deal with the scenario of the record being in the form of `foo.example.com`? That is, full enough to allow for the zone lookup, but still a relative name.

I think it would make much more sense to depend on that the full name is an absolute name in the form of `foo.example.com.`. Not only will that work out of the box without any shortening, it's also much more in line with how the nsupdate module works, as well as with general DNS conventions.

Preferably there should be a separate previous check which ensure that when the zone parameter is None then the record parameter must be an absolute name. It could be something as simple as requiring that the record name ends in a dot.

Also, the current shortening actually breaks absolute record names, turning `foo.example.com.` into `foo.`.